modules/common/instance/instance.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instance
+
+import (
+	"context"
+
+	helper "github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	util "github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// PatchInstance - Patch an instance's metadata and/or status if they have changed (based
+// on comparison with its old state as represented in the helper.Helper)
+//
+// NOTE: This function is mainly intended for use in Podified Operators with their
+// deferred-instance-persistence pattern within the reconcile loop.  If used in this manner,
+// any error returned by this function should be set as the error returned by the encompassing
+// Reconcile(...) function so that the error from PatchInstance is properly propagated.
+//
+// Example:
+//
+// func (r *SomeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, _err error) {
+//     ...
+//     defer func() {
+//         ...
+//         err := instance.PatchInstance(ctx, h, instance)
+//         if err != nil {
+//             _err = err
+//             return
+//         }
+//     }
+//     ...
+// }
+//
+func PatchInstance(ctx context.Context, h *helper.Helper, instance client.Object) error {
+	var err error
+
+	if err = h.SetAfter(instance); err != nil {
+		util.LogErrorForObject(h, err, "Set after and calc patch/diff", instance)
+		return err
+	}
+
+	changes := h.GetChanges()
+	patch := client.MergeFrom(h.GetBeforeObject())
+
+	if changes["metadata"] {
+		err = h.GetClient().Patch(ctx, instance, patch)
+		if k8s_errors.IsConflict(err) {
+			util.LogForObject(h, "Metadata update conflict", instance)
```

Comment:
Lets change this to
```golang
l := log.FromContext(ctx)
l.Info(...) 
```
The instance related information is already stored in and emitted by that logger